lib/elixir/lib/exception.ex

```diff
@@ -683,6 +691,10 @@ defmodule UndefinedFunctionError do
     "      * " <> fun <> "/" <> Integer.to_string(arity) <> "\n"
   end
 
+  defp macro_with_same_mfa?(module, function, arity) do
+    function_exported?(module, :__info__, 1) and macro_exported?(module, function, arity)
```

Comment:
> Me and @lexmag quickly discussed about this and maybe it would make sense to handle Erlang modules in macro_exported?/3 and return false for them

Yes, we should fix it.